azurerm/resource_arm_iothub.go

```diff
@@ -114,12 +114,12 @@ func resourceArmIotHub() *schema.Resource {
 
 			"shared_access_policy": {
 				Type:     schema.TypeList,
-				Computed: true,
+				Optional: true,
```

Comment:
this is going to need to become:

```suggestion
				Type: schema.TypeSet,
				Optional: true,
				Computed: true,
```

This needs to remain `computed` since it's possible that users may only want to retrieve these and not set the values; otherwise this'll be a breaking change.

In addition - since these can now be defined in-line this'll need to be a `TypeSet`, which is a breaking change - but is required since these could be returned in any order at this point (and may differ from the ordering defined in a users config). Unfortunately this would also be a breaking change; as such I believe defining these in-line may not be the best way to expose these fields.

Instead I believe it would make sense to expose a Virtual Resource (ala the azurerm_virtual_Machine_data_disk_attachment` resource which looks up the VM and adds the Disk, then returns) as an independent resource - rather than defining these in-line. This would also allow these to be retrieved as a Data Source.

Unfortunately this would require the use of a Lock to ensure concurrent changes aren't made; however it should be possible to do so - as it appears there's no independent API available to manage Shared Access Policies within an IoTHub.

What do you think?